Update vaapi transcoding (h264/hevc)

ovaron76github opened the pull request:
Buffer factor (buffersize manipulation) now configurable at WebUI.
Added rc_mode (h264/hevc) and tier (hevc), both configurable via WebUI.
Removed B-Frame workaround from the past at HEVC.

ovaron76github commented:
Can this pull request be merged with this project or is there some error which I should fix before?
On my side it works without problems.

Flole998 commented on src/transcoding/codec/codecs/libs/vaapi.c:
```diff
@@ -238,16 +266,21 @@ tvh_codec_profile_vaapi_hevc_open(tvh_codec_profile_vaapi_t *self,
     // max_b_frames
     // XXX: remove when b-frames handling in vaapi_encode is fixed
-    AV_DICT_SET_INT(opts, "bf", 0, 0);
+    //AV_DICT_SET_INT(opts, "bf", 0, 0);  //seems to be fixed
```
Please remove this completely if you are under the impression that it's no longer needed. Otherwise it'll be forgotten soon and this commented out code will stay here forever.

ovaron76github replied:
I will close this request and create a new one.
In forum someone asked if it is possible to directly add -vf parameters and I already added a field to vaapi.
I need to wait until I get some response if it's working and after this I close this request and start a new one with the updated code.

ovaron76github commented:
Added an option for ignore b-frames because some drivers needs this (AMD).
Also updated VP8 and VP9 encoding. 
VP8 and VP9 now also reacts on options for "Buffer factor" and "Ignore B-Frames" and got the "force_key_frames" expression to increase picture quality.

Flole998 commented:
Can you please squash your commits together? They should appear as a single commit with a meaningful commit description.
